v3/internal/runtime/desktop/@wailsio/runtime/src/cancellable.ts

```
@@ -0,0 +1,944 @@
+        Object.assign(pwr, promiseWithResolvers<void>());
+    }
+    if (promise[barrierSym] == null) {
+        pwr.resolve!();
+        promise[barrierSym] = pwr;
+    }
+    return pwr.promise!;
+}
+
+// Stop sneaky people from breaking the barrier mechanism.
+const promiseThen = Promise.prototype.then;
+Promise.prototype.then = function(...args) {
```

Biome (1.9.4)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)